### ISSUE TITLE
Don't understand how to use NLog.Extended

### ISSUE THREAD

**Issue by cvalde** [NLog/NLog#2826]
Hello, using VS2015, .net 4.5, NLog.Extended 4.5.7.
I thought naively that using a line like this in Nlog.config:

```xml
<variable name="logFilePath" value="C:\temp\NLog\CoreSmartDoctorCode-${appsetting:name=sufijo:default=null}-${shortdate}.log"/>
```
and adding this line to web.config
```xml
<add key="sufijo" value="Rest" />
```
would replace` ${appsetting:name=sufijo:default=null}` by the value `"Rest`" but it gets replaced by an empty string, so the variable "logFilePath" becomes
`CoreSmartDoctorCode--2018-07-31.log`

Notice there's nothing between the two dashes. As this is the first time I try Nlog.Extended, I should be doing something obvious wrong. Can you please tell me what's wrong?

Thanks.
Claudio.

However, it generates an empty string

**Comment by snakefoot** [comment]
Any warnings / errors when using the internal logger: https://github.com/NLog/NLog/wiki/Internal-Logging

**Comment by cvalde** [comment]
Hello, thanks for your prompt answer. I think this line is important:

2018-08-01 09:53:29.9573 Error Error parsing layout appsetting will be ignored. Exception: System.ArgumentException: LayoutRenderer cannot be found: 'appsetting'
   en NLog.Config.Factory`2.CreateInstance(String name)
   en NLog.Layouts.LayoutParser.GetLayoutRenderer(ConfigurationItemFactory configurationItemFactory, String name)

In other post, I see that this part was added to NLog.log:
<extensions>
<add assembly="NLog.WindowsEventLog" />
</extensions>

Therefore, I tried this:

  <extensions>
    <add assembly="NLog.Extended"/>
  </extensions>

However, this made the problem worse:
2018-08-01 10:08:32.1623 Info Loading assembly: NLog.Extended
2018-08-01 10:08:32.5143 Error Error loading extensions. Exception: System.IO.FileNotFoundException: No se puede cargar el archivo o ensamblado 'NLog.Extended' ni una de sus dependencias. El sistema no puede encontrar el archivo especificado.
Nombre de archivo: 'NLog.Extended'
According to NuGet, I have NLog 4.5.7, NLog.Config 4.5.7, NLog.Extended 4.5.7 and NLog.Schema 4.5.7 installed.

Not sure how to continue. Thanks for your patience, I never tried to use extensions before.
Claudio.

**Comment by snakefoot** [comment]
@cvalde NLog.Extended depends on Msmq, so you probably need that dependency for it to work.

I would probably write a custom layout renderer that can lookup appsettings, instead of using NLog.Extended:

https://github.com/NLog/NLog/wiki/How-to-write-a-custom-layout-renderer

At least until this has been resolved: #2787

**Comment by 304NotModified** [comment]
I assume your question has been answered, if not, please let us know!
